tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java

```diff
@@ -75,20 +86,44 @@ public void testRfc2617() throws Exception {
         Assert.assertEquals(UNAUTHORIZED, response.getStatusCode());
         Assert.assertEquals("Digest realm=\"testrealm@host.com\", nonce=\"AAAAAQABsxiWa25/kpFxsPCrpDCFsjkTzs/Xr7RPsi/VVN6faYp21Hia3h4=\", opaque=\"00000000000000000000000000000000\", algorithm=MD5, qop=auth", response.getAuthenticateHeader());
 
-        TestingHttpServerRequest request2 = new TestingHttpServerRequest(new String[] {
+        testStatefulDigest(new String[] {
                 "Digest username=\"Mufasa\",\n" +
-                "                 realm=\"testrealm@host.com\",\n" +
-                "                 nonce=\"dcd98b7102dd2f0e8b11d0f600bfb0c093\",\n" +
-                "                 uri=\"/dir/index.html\",\n" +
-                "                 qop=auth,\n" +
-                "                 nc=00000001,\n" +
-                "                 cnonce=\"0a4f113b\",\n" +
-                "                 response=\"6629fae49393a05397450978507c4ef1\",\n" +
-                "                 opaque=\"00000000000000000000000000000000\",\n" +
-                "                 algorithm=MD5"
-        });
-        mechanism.evaluateRequest(request2);
-        Assert.assertEquals(Status.COMPLETE, request2.getResult());
+                        "                 realm=\"testrealm@host.com\",\n" +
+                        "                 nonce=\"dcd98b7102dd2f0e8b11d0f600bfb0c093\",\n" +
+                        "                 uri=\"/dir/index.html\",\n" +
+                        "                 qop=auth,\n" +
+                        "                 nc=00000001,\n" +
+                        "                 cnonce=\"0a4f113b\",\n" +
+                        "                 response=\"6629fae49393a05397450978507c4ef1\",\n" +
+                        "                 opaque=\"00000000000000000000000000000000\",\n" +
+                        "                 algorithm=MD5"
+        },mechanism);
+
+        // test case insensitive
+        testStatefulDigest(new String[] {
+                "digest username=\"Mufasa\",\n" +
+                        "                 realm=\"testrealm@host.com\",\n" +
+                        "                 nonce=\"dcd98b7102dd2f0e8b11d0f600bfb0c093\",\n" +
+                        "                 uri=\"/dir/index.html\",\n" +
+                        "                 qop=auth,\n" +
+                        "                 nc=00000001,\n" +
+                        "                 cnonce=\"0a4f113b\",\n" +
+                        "                 response=\"6629fae49393a05397450978507c4ef1\",\n" +
+                        "                 opaque=\"00000000000000000000000000000000\",\n" +
+                        "                 algorithm=MD5"
+        },mechanism);
+        testStatefulDigest(new String[] {
+                "DIGEST username=\"Mufasa\",\n" +
```

Comment:
Hi @keshav-redhat , looks good! Just a nitpick, since we have tests that use "Digest" case successfully, a single call with different case should be sufficient to test case sensitivity.  Either "DIGEST" or "digest" or some combination should suffice:

```
// test case insensitive
        testStatefulDigest(new String[] {
                "dIgEsT username=\"Mufasa\",\n" +
                        "                 realm=\"testrealm@host.com\",\n" +
                        "                 nonce=\"dcd98b7102dd2f0e8b11d0f600bfb0c093\",\n" +
                        "                 uri=\"/dir/index.html\",\n" +
                        "                 qop=auth,\n" +
                        "                 nc=00000001,\n" +
                        "                 cnonce=\"0a4f113b\",\n" +
                        "                 response=\"6629fae49393a05397450978507c4ef1\",\n" +
                        "                 opaque=\"00000000000000000000000000000000\",\n" +
                        "                 algorithm=MD5"
        },mechanism);
```